### DISCUSSION TITLE
"brew upgrade" always ends in error

### DISCUSSION THREAD

**Discussion by manroygood** [Homebrew/discussions#1629] [Everyday usage]
Every time I use the command `brew upgrade` it ends after installing one formula and I have to run it again. I always get an error like this:

```
==> `brew cleanup` has not been run in 30 days, running now...
Removing: /usr/local/Cellar/sphinx-doc/3.4.2... (3,853 files, 55.3MB)
Error: Permission denied @ apply2files - /usr/local/share/ghostscript/9.07/Resource/Font/blex.pfb
```
Then I run `brew upgrade` again and it upgrades the next formula in the list and exits with a similar error.

**Comment by SMillerDev** [comment]
Somehow your user is not allowed to edit `/usr/local/share/ghostscript/9.07/Resource/Font/blex.pfb`, you can run `sudo chown $(whoami) /usr/local/share/ghostscript/9.07/Resource/Font/blex.pfb` to claim ownership again.

**Reply by manroygood** [reply]
Okay, but there seems to be a whole class of such files whose ownership is set incorrectly. I now run `brew upgrade` again and now it moves on to upgrade ruby and it exits with the error message:

```
==> `brew cleanup` has not been run in 30 days, running now...
Removing: /usr/local/Cellar/ruby/3.0.0_1... (16,314 files, 37.6MB)
Error: Permission denied @ apply2files - /usr/local/share/ghostscript/9.07/Resource/Font/blex.pfb
```

again, it fails to move on to the next formula needing to be upgraded.

**Reply by Paul-Morris** [reply]
Are the files in /usr/local set to root ownership? Mine were and I got similar errors as you. I fixed it by changing to my user with this but I don't know if it is recommended (I was desperate):

**sudo chown -R $(whoami) /usr/local/***

**Reply by manroygood** [reply]
That worked. instead of `usr/local` I chowned `usr/local/share` since all the errors seemed to involve files in this subdirectory.

**Reply by Paul-Morris** [reply]
> That worked. instead of `usr/local` I chowned `usr/local/share` since all the errors seemed to involve files in this subdirectory.

Great good to hear!
